### DISCUSSION TITLE
Custom edge does not fire addSelectedNodes when user clicks on edge

### DISCUSSION THREAD

**Discussion by ACmiester** [xyflow/xyflow#3001] [Q&A]
Hey,

Version:11.5.6,

I have created a custom edge where the user can click on the edge, and then it should select the nodes it is connected to. Therefore we are using the `addSelectedNodes` to select these nodes, however, the issue is that the nodes does not get selected. It seems that the function works elsewhere but not in my custom edge. Something i'm doing wrong?

Below is a piece of the component, and it should be noted that there is 4 nodes, "a", "b", "c" and "d".

Edge.tsx
```const AbstractEdge = ({
  id,
  data,
  selected,
}: EdgeProps<DefaultEdgeData>) => {
  const selectNodes = useSelectNodes();

  const onSelectEdge = () => {
    selectNodes(['b', 'c']);
  };

  return (
    <>
       ...
    </>
  );
};

export default memo(AbstractEdge);
```

useSelectNodes.ts
```
export const useSelectNodes = () => {
  const selectNodes = useStore(selectNodesSelector);
  return selectNodes;
};

const selectNodesSelector = (state: ReactFlowState) => state.addSelectedNodes;
```

**Comment by moklick** [comment]
lgtm, could you create a codesandbox https://new.reactflow.dev/react-js
